src/index.ts

```diff
@@ -686,12 +686,32 @@ function getPublicKeyFor<T extends MessageTypes>(
   return recoverPublicKey(msgHash, msgParams.sig);
 }
 
-function padWithZeroes(number: string, length: number): string {
-  let myString = `${number}`;
-  while (myString.length < length) {
-    myString = `0${myString}`;
```

Comment:
This computes a new string and stores it in memory `length - number.length` times.